Libs/MRML/DisplayableManager/vtkMRMLSliceIntersectionDisplayableManager.cxx

```diff
@@ -0,0 +1,270 @@
+    this->SliceIntersectionInteractionWidget->CreateDefaultRepresentation();
+    this->SliceIntersectionInteractionWidget->SetRenderer(this->External->GetRenderer());
+    this->SliceIntersectionInteractionWidget->SetSliceNode(this->GetSliceNode());
+    }
+  else
+    {
+    this->SliceIntersectionInteractionWidget->SetSliceNode(this->GetSliceNode());
+    }
+}
+
+//---------------------------------------------------------------------------
+// vtkMRMLSliceIntersectionDisplayableManager methods
+
+//---------------------------------------------------------------------------
+vtkMRMLSliceIntersectionDisplayableManager::vtkMRMLSliceIntersectionDisplayableManager()
+{
+  this->Internal = new vtkInternal(this);
+}
+
+//---------------------------------------------------------------------------
+vtkMRMLSliceIntersectionDisplayableManager::~vtkMRMLSliceIntersectionDisplayableManager()
+{
+  delete this->Internal;
+}
+
+//---------------------------------------------------------------------------
+void vtkMRMLSliceIntersectionDisplayableManager::PrintSelf(ostream& os, vtkIndent indent)
+{
+  this->Superclass::PrintSelf(os, indent);
+}
+
+//---------------------------------------------------------------------------
+void vtkMRMLSliceIntersectionDisplayableManager::ObserveMRMLScene()
+{
+  this->Superclass::ObserveMRMLScene();
+
+  // Observe all the slice composite nodes in the scene to trigger rendering when the interactive slice intersection
+  // is turned on or off.
+  //TODO: This is a hack, and should be done differently. The reasons this workaround was applied are:
+  // - There is no way to get the composite node for a given slice node without the slice logic, which cannot be accessed from here
+  // - The composite node is known from the widget representation, but the RequestRender function in this class seems to be the only
+  //   way to get re-render, and there is no way to trigger calling that function from the representation
+  // - Note: The reason update works in the traditional slice intersection is that the slice model node is modified in that case,
+  //   and the model slice displayable manager handles that and calls RequestRender. But for the interactive slice intersection the
+  //   slice model is not modified.
+  // ! A possible way to make this nicer (and not observe three nodes instead of one) could be to get the slice composite node
+  //   in vtkMRMLSliceIntersectionInteractionRepresentation::SetSliceNode, get the corresponding slice node there, and observe
+  //   the composite node, which would call a Modified on the slice node, which in turn can be observed from this class directly.
+  vtkEventBroker* broker = vtkEventBroker::GetInstance();
+  vtkMRMLScene* scene = this->GetMRMLScene();
+  std::vector<vtkMRMLNode*> sliceCompositeNodes;
+  int numOfSliceCompositeNodes = scene->GetNodesByClass("vtkMRMLSliceCompositeNode", sliceCompositeNodes);
```

Comment:
I think simply that the original slice intersection flag was stored there.

Reply:
I do not know the reason why the slice intersection visibility flag was originally stored in the composite node. It was strange to me when I started working on this. After identifying that, I decided to keep the flags related to the interactive slice intersection feature also in the composite node.

See: https://github.com/Slicer/Slicer/blob/master/Libs/MRML/Core/vtkMRMLSliceCompositeNode.cxx#L53

Reply:
It is good that you followed the existing pattern, but if it leads to simpler code (for example less observers) then we should move all these options to the slice node. We may even decide to move everything to the slice node even if it does not make anything particularly simpler, because adding a new feature is a good time to refactor the code to make things right.